common/scala/src/main/scala/whisk/core/containerpool/logging/ElasticSearchRestClient.scala

```diff
@@ -108,7 +109,8 @@ object ElasticSearchJsonProtocol extends DefaultJsonProtocol {
   implicit object EsQueryMustJsonFormat extends RootJsonFormat[EsQueryMust] {
     def read(query: JsValue) = ???
     def write(query: EsQueryMust) = {
-      val boolQuery = Map("must" -> query.matches.toJson) ++ query.range.map(r => "filter" -> r.toJson)
+      val boolQuery = Map("must" -> query.matches.toJson) ++ Map("filter" -> query.range.toJson)
+        .filter(_._2 != JsArray())
```

Comment:
Does it make sense to change this to `.filter(_._2.elements.nonEmpty)` to make it more clear? Do we even need to filter this out? (I said this in the other comment, but maybe it's not necessary)?

Reply:
I'd rather not send `filter: {}` when there are no ranges provided.

Reply:
`filter(_._2 != JsArray.empty)` :)

Reply:
`.filter(_._2.elements.nonEmpty)` doesn't work because the map's value is an JsValue instead of a JsArray.